cpp/include/cudf/types.hpp

```diff
@@ -191,6 +195,16 @@ enum type_id {
   TIMESTAMP_MILLISECONDS,  ///< duration of milliseconds since Unix Epoch in int64
   TIMESTAMP_MICROSECONDS,  ///< duration of microseconds since Unix Epoch in int64
   TIMESTAMP_NANOSECONDS,   ///< duration of nanoseconds since Unix Epoch in int64
+  DURATION_YEARS,          ///< duration of years in int32
```

Comment:
We need days since we have `TIMESTAMP_DAYS`, but otherwise agreed for now.

In the future the goal would be to support most of these though 😢

Reply:
YEARS/MONTHS/WEEKS/DAYS/MINUTES would be useful for constructing all timedelta operations. This would allow  for eg. (+1 year + 1 month +1 days) operations with timestamp. 
This is still open for discussion.

Reply:
> useful for constructing all timedelta operations

Useful, yes, practical, no. I do not think our compile time or library file size can handle add all of these types in addition to all of the other new ones being added.

Reply:
FWIW, Arrow only supports durations on seconds, milliseconds, microseconds, and nanoseconds.

Numpy supports it on years, months, weeks, days, hour, and minutes as well but uses 64 bit types for all of them.

Maybe we only support seconds, milliseconds, microseconds, and nanoseconds for now, considering using a duration of 64 bit seconds allows you to capture all of a duration of 32 bit days, albeit with memory / performance tradeoffs.

Reply:
I'm definitely in favor of supporting the minimal set of essential types.

Reply:
For spark we need MONTHS, DAYS, and MICROSECONDS.

This is because if I want to go one year in the future it is 12 months, but it is not always 365 days.  If I want to go 1 week in the future it is always 7 days, but I cannot always say how many microseconds it is.

Reply:
Well, looks like `std::chrono::duration` is all just ratio based: https://en.cppreference.com/w/cpp/chrono/duration

| Unit | Ratio |
| - | - |
| seconds | 1 |
| minutes | 60 |
| hours | 3600 |
| days | 86400 |
| weeks | 604800 |
| months | 2629746 |
| years | 31556952 |

Reply:
> This is because if I want to go one year in the future it is 12 months, but it is not always 365 days.

According to C++ duration types, a year is always 365.2425 days. See:https://en.cppreference.com/w/cpp/chrono/duration

>  years is equal to 365.2425 days (the average length of a Gregorian year). months is equal to 30.436875 days (exactly 1/12 of years).

Reply:
now, DAYS, SECONDS, MILLISECONDS, MICROSECONDS, NANOSECONDS are the only types in duration type.